src/parser/jobs/sam/modules/Meikyo.tsx

```diff
@@ -58,4 +64,27 @@ considerAction(action: Action) {
 	return false
 
 }
+
+// override for end of fight reducing
+
+reduceExpectedGCDsEndOfFight(buffWindow: BuffWindowState): number  {
+		if ( this.buffStatus.duration ) {
+			// Check to see if this window is rushing due to end of fight - reduce expected GCDs accordingly
+			const windowDurationMillis = this.buffStatus.duration * 1000
+			const fightTimeRemaining = this.parser.pull.duration - (buffWindow.start - this.parser.eventTimeOffset)
+
+			if (windowDurationMillis >= fightTimeRemaining) {
+				// This is using floor instead of ceiling to grant some forgiveness to first weave slot casts at the cost of 2nd weaves might be too forgiven
+				const possibleGCDs = Math.floor(fightTimeRemaining / SAM_BASE_GCD_SPEED_BUFFED)
+
+				if (possibleGCDs < SEN_GCDS) {
+					const reduceGCDsBy = (SEN_GCDS - possibleGCDs)
+					return reduceGCDsBy
+				}
+			}
+		}
```

Comment:
Kinda over complicated imo. Why not initialise like `let reduceGCDsBy = 0`, always return it, and modify it, or just clamp or `Math.max` the value.

```
let reduceGCDsBy = 0

if (windowDurationMillis >= fightTimeRemaining) {
  const possibleGCDs = Math.floor(fightTimeRemaining / SAM_BASE_GCD_SPEED_BUFFED)

  if (possibleGCDs < SEN_GCDS) {
    reduceGCDsBy -= (SEN_GCDS - possibleGCDs)
  }
}

return reduceGCDsBy
```

or even like, 
```
if (windowDurationMillis >= fightTimeRemaining) {
  const possibleGCDs = Math.floor(fightTimeRemaining / SAM_BASE_GCD_SPEED_BUFFED)

  return Math.max((SEN_GCDS - possibleGCDs), 0)
}

return 0
```

Reply:
I think the latter was cleaner fwiw but yolo